src/dash-table/derived/cell/dropdowns.ts

```diff
@@ -1,94 +1,128 @@
 import * as R from 'ramda';
 
-import { memoizeOneFactory } from 'core/memoizer';
+import { memoizeOne } from 'core/memoizer';
+import memoizerCache from 'core/cache/memoizer';
+import SyntaxTree from 'core/syntax-tree';
+
+import {
+    IConditionalDropdown
+} from 'dash-table/components/CellDropdown/types';
 
 import {
     Data,
     Datum,
     VisibleColumns,
     ColumnId,
     Indices,
-    IColumnDropdown,
-    IConditionalColumnDropdown,
-    IDropdownProperties,
-    DropdownValues
+    DropdownValues,
+    IBaseVisibleColumn,
+    IVisibleColumn
 } from 'dash-table/components/Table/props';
-import SyntaxTree from 'core/syntax-tree';
-import memoizerCache from 'core/memoizerCache';
-import { IConditionalDropdown } from 'dash-table/components/CellDropdown/types';
 
 const mapData = R.addIndex<Datum, (DropdownValues | undefined)[]>(R.map);
 
-const getDropdown = (
-    astCache: (key: [ColumnId, number], query: string) => SyntaxTree,
-    conditionalDropdowns: IConditionalDropdown[],
-    datum: Datum,
-    property: ColumnId,
-    staticDropdown: DropdownValues | undefined
-): DropdownValues | undefined => {
-    const dropdowns = [
-        ...(staticDropdown ? [staticDropdown] : []),
-        ...R.map(
-            ([cd]) => cd.dropdown,
-            R.filter(
-                ([cd, i]) => astCache([property, i], cd.condition).evaluate(datum),
-                R.addIndex<IConditionalDropdown, [IConditionalDropdown, number]>(R.map)(
-                    (cd, i) => [cd, i],
-                    conditionalDropdowns
-                ))
-        )
-    ];
+export default () => new Dropdowns().get;
 
-    return dropdowns.length ? dropdowns.slice(-1)[0] : undefined;
-};
+class Dropdowns {
+    /**
+     * Return the dropdown for each cell in the table.
+     */
+    get = memoizeOne((
+        columns: VisibleColumns,
+        data: Data,
+        indices: Indices,
+        columnConditionalDropdown: any,
+        columnStaticDropdown: any,
+        dropdown_properties: any
+    ) => mapData((datum, rowIndex) => R.map(column => {
+        const applicable = this.applicable.get(column.id, rowIndex)(
+            column,
+            indices[rowIndex],
+            columnConditionalDropdown,
+            columnStaticDropdown,
+            dropdown_properties
+        );
 
-const getter = (
-    astCache: (key: [ColumnId, number], query: string) => SyntaxTree,
-    columns: VisibleColumns,
-    data: Data,
-    indices: Indices,
-    columnConditionalDropdown: IConditionalColumnDropdown[],
-    columnStaticDropdown: IColumnDropdown[],
-    dropdown_properties: IDropdownProperties
-): (DropdownValues | undefined)[][] => mapData((datum, rowIndex) => R.map(column => {
-    const realIndex = indices[rowIndex];
+        return this.dropdown.get(column.id, rowIndex)(
+            applicable,
+            column,
+            datum
+        );
+    }, columns), data));
```

Comment:
First level -- in many case the table get regenerated because the active cell has changed -- this does not affect the dropdowns -- only regenerate if the table structure, data or dropdown options have changed.